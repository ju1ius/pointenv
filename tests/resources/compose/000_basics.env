# basic syntax as documented in:
# https://docs.docker.com/compose/environment-variables/env-file/#syntax

EMPTY=
UNQUOTED=foo bar=baz $"a"$'b'
SINGLE_QUOTED='foo $bar \' baz'
DOUBLE_QUOTED="foo \$bar $ $'foo' \" baz"

D_ESCAPED="A\n\r\tB\\C\X"
S_ESCAPED='A\n\r\tB\\C'

# backslash at end should not trigger line continuation
U_ESCAPED=a\b\
FOO=foo
