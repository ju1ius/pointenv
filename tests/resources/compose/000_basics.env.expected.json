[
  {
    "key": "DOUBLE_QUOTED",
    "value": "foo $bar $ $'foo' \" baz"
  },
  {
    "key": "D_ESCAPED",
    "value": "A\n\r\tB\\C\\X"
  },
  {
    "key": "EMPTY",
    "value": ""
  },
  {
    "key": "FOO",
    "value": "foo"
  },
  {
    "key": "SINGLE_QUOTED",
    "value": "foo $bar \\' baz"
  },
  {
    "key": "S_ESCAPED",
    "value": "A\\n\\r\\tB\\\\C"
  },
  {
    "key": "UNQUOTED",
    "value": "foo bar=baz $\"a\"$'b'"
  },
  {
    "key": "U_ESCAPED",
    "value": "a\\b\\"
  }
]
